components/Starknet/modules/architecture-and-concepts/pages/smart-contracts/system-calls-cairo1.adoc

```diff
@@ -463,3 +463,68 @@ None.
 .Common library
 
 link:https://github.com/starkware-libs/cairo/blob/cca08c898f0eb3e58797674f20994df0ba641983/corelib/src/starknet/syscalls.cairo#L70[`syscalls.cairo`^]
+
+[id="keccak"]
+== `keccak`
+
+[discrete]
+=== Function signature
+
+[source,cairo,subs="+quotes,+macros"]
+----
+extern fn keccak_syscall(
+    data: Span<felt252>
+) -> SyscallResult<u256> implicits(GasBuiltin, System) nopanic;
+----
+
+[discrete]
+=== Description
+
+Computes the Keccak-256 hash of the input data. This system call is particularly useful when interacting with Ethereum contracts or implementing Ethereum-compatible functionality, as Keccak-256 is widely used in the Ethereum ecosystem.
+
+[discrete]
+=== Arguments
+
+[horizontal,labelwidth=35]
+`_data_: Span<felt252>`:: The input data to be hashed, represented as an array of field elements.
+
+[discrete]
+=== Return values
+
+* The Keccak-256 hash of the input data as a `u256` value, wrapped in `SyscallResult`.
+
+[discrete]
+=== Common library
+
+link:https://github.com/starkware-libs/cairo/blob/cca08c898f0eb3e58797674f20994df0ba641983/corelib/src/starknet/syscalls.cairo#L84[`syscalls.cairo`^]
+
+[discrete]
+=== Example
+
+The following example computes the Keccak-256 hash of an array containing two elements:
+
+[source,cairo]
+----
+use array::ArrayTrait;
+use starknet::syscalls::keccak_syscall;
+
+// Create input data
+let mut data = ArrayTrait::new();
+data.append(1);
+data.append(2);
+
+// Compute Keccak hash
+let hash = keccak_syscall(data.span()).unwrap_syscall();
+----
+
+[NOTE]
```

Comment:
This note just reiterates stuff that has already been mentioned (unless I'm missing something), so please delete it.